tests/panoptic/test_panoptic_utils.py

```diff
@@ -0,0 +1,357 @@
+    assert label2id == {"MOTORBOAT": 0, "SPHERICAL_BUOY": 1}
+    assert (pred == gt).all()
+    assert np.unique(pred).size == 3
+    assert np.unique(gt).size == 3
+
+
+def test_payload_to_seg_metric_empty_payload():
+    payload = {}
+    model_name = "model"
+    label2id = {"WATER": 0, "SKY": 1}
+
+    with pytest.raises(AttributeError):
+        _, _, label2id = payload_to_seg_metric(payload, model_name, label2id)
+
+    assert label2id == {"WATER": 0, "SKY": 1}
+
+
+def test_payload_to_seg_metric_missing_model_name():
+    payload = generate_synthetic_payload(gt_config=[1], model_config=[1])
+    with pytest.raises(TypeError):
+        payload_to_seg_metric(payload=payload)
+
+
+def test_payload_to_seg_metric_model_name_not_on_sequence():
+    payload = generate_synthetic_payload(gt_config=[1], model_config=[1])
+    with pytest.raises(AttributeError):
+        payload_to_seg_metric(payload=payload, model_name="not_model")
+
+
+def test_payload_to_seg_metric_missing_gt_field():
+    payload = generate_synthetic_payload(
+        gt_field_name=False, gt_config=[1], model_config=[1]
+    )
+    with pytest.raises(AttributeError):
+        payload_to_seg_metric(payload=payload, model_name="model")
+
+
+def test_payload_to_seg_metric_label2id_populated():
+    payload = generate_synthetic_payload(gt_config=[1], model_config=[2])
+    label2id = {"WATER": 0, "SKY": 1}
+    _, _, label2id = payload_to_seg_metric(
+        payload=payload, model_name="model", label2id=label2id
+    )
+    assert label2id == {'WATER': 0, 'SKY': 1, 'MOTORBOAT': 2, 'SPHERICAL_BUOY': 3}
+
+
+def test_payload_to_seg_metric_none_label2id():
+    payload = generate_synthetic_payload(gt_config=[1], model_config=[2])
+    label2id = None
+    pred, gt, label2id = payload_to_seg_metric(
+        payload=payload, model_name="model", label2id=label2id
+    )
+    assert label2id == {"MOTORBOAT": 0, "SPHERICAL_BUOY": 1}
+
+def test_payload_to_seg_metric_single_frame():
+    payload = generate_synthetic_payload(gt_config=[1], model_config=[1])
+    pred, gt, label2id = payload_to_seg_metric(payload=payload, model_name="model")
+    assert label2id == {'MOTORBOAT': 0}
+    assert pred.shape == (1, 512, 640, 2)
+    assert gt.shape == (1, 512, 640, 2)
+    assert (pred == gt).all()
+    assert np.unique(pred).size == 2
+    assert np.unique(gt).size == 2
+
+
+def test_payload_to_seg_metric_partial_gt_and_preds():
+    payload = generate_synthetic_payload(gt_config=[1, 2, 0], model_config=[1, 0, 2])
+    pred, gt, label2id = payload_to_seg_metric(payload=payload, model_name="model")
+    assert label2id == {'MOTORBOAT': 0, 'SPHERICAL_BUOY': 1}
+    assert pred.shape == (3, 512, 640, 2)
+    assert gt.shape == (3, 512, 640, 2)
+    assert (pred != gt).any()
+    assert pred[0,:,:,0].sum() == pred[0,:,:,1].sum() == -327630
+    assert pred[1,:,:,0].sum() == pred[1,:,:,1].sum() == -327680
+    assert pred[2,:,:,0].sum() == pred[2,:,:,1].sum() == -327560
```

Comment:
I do not get how this number is created (but it should be correct as the tests are already checked in the PR, no?). i would expect it to be 327680-85